lib/tether/tether.go

```diff
@@ -339,6 +340,8 @@ func (t *tether) processSessions() error {
 					err:  t.launch(session),
 				}
 			}(session)
+
+			session.Unlock()
```

Comment:
You have this inside the `if {...}` clause only. If this clause is skipped, you will deadlock at the start of the loop when you try to lock again.